erp-client/src/app/components/award/createAward.component.html

```diff
@@ -0,0 +1,83 @@
+<mat-toolbar>Create Award</mat-toolbar>
+
+<div class="container">
+  <mat-card>
+    <mat-card-header>
+      <!-- <mat-card-title></mat-card-title> -->
+      <mat-card-subtitle>
+        <p>Use the form below to create a new award for one of the Employees.</p>
+      </mat-card-subtitle>
+    </mat-card-header>
+
+    <form [formGroup]="createAwardForm" (ngSubmit)="onSubmit()">
+      <mat-card-content>
+        <mat-form-field>
+          <input matInput type="text" placeholder="ID" formControlName="id" [ngClass]="{ 'is-invalid': submitted && f.id.errors }" />
+        </mat-form-field>
+
+        <mat-form-field>
+          <!-- <input matInput type="text" placeholder="Created By" formControlName="user" [ngClass]="{ 'is-invalid': submitted && f.user.errors }" />
+                      <mat-error *ngIf="submitted && f.user.errors" class="invalid-feedback">
+                          <mat-error *ngIf="f.user.errors.required">User is required</mat-error>
+                      </mat-error> -->
+          <mat-select (selectionChange)='onUserSelectChange()' placeholder="Created By" formControlName="user" [ngClass]="{ 'is-invalid': submitted && f.user.errors }">
```

Comment:
Right now, this uses a <select>, since the functionality to know which User is creating the Award under the hood isn't yet there. Once we get that part figured out, I figure we can either change it to a disabled input field (if we want the user to still be able to see this info (which may be nice for the person grading our work) --- or we can make the input hidden all together.

Reply:
Seems fine for now. I think we should aim for this part to be handled on the server but works for testing.

Reply:
That's a good point. I'm not sure why I didn't think of that. Makes sense that if we're already logging everything, it'd be redundant to include the user again in the award table.

As a counter-point to my own thought above --- I know we need to have the functionality for a user to to "Delete from the system awards that this user has previously given." So we'll need to make sure we can reasonable fetch all awards that the logged-in user created. Of course, we can dig through the log table to:
> Just brainstorming here
1. Get all the log entries that are related to awards where the modified_by user is the logged-in user
2. Then get only the distinct award.ids from the results of step 1.
3. Select all awards from the results of step 2.

On the other hand, if we store the user_account.id in the award table, it introduces a bit of redundancy ```RIP Boyce Codd```, then we have the trade-off of just being able to do a regular ol'
```sql
select * from award where user_account.id = ?
```

Not sure which is the better route.

Reply:
Totally, definitely trade-offs here. Even if we decide to keep the user account id field in the award table I think _eventually_ we'd still want to get the current user at the controller level and handle that logic in the server.

This is definitely helpful though in its current state for testing in the sense that the user generating the award is selectable, so this definitely shouldn't block your merge. I'm just sort of thinking out loud.

Reply:
Right on -- and same. I'm permanently down with doing whatever is better for the squad.